client_test/helpers_test.go

```diff
@@ -34,42 +36,39 @@ const PERSIST_STORE_FOLDER = "../data/client_test"
 // If the timeout lapses and the objectives have not all completed, the parent test will be failed.
 func waitTimeForCompletedObjectiveIds(t *testing.T, client *client.Client, timeout time.Duration, ids ...protocols.ObjectiveId) {
```

Comment:
As discussed in the commit messages, it would not be difficult to entirely deprecate this function. 

While I think it is a good function, it constitutes quite some complexity and "cleverness". If we want to keep it, it would be better to push _inside_ the production code. 

Complexity in test code implies complexity for the API consumer.

Reply:
I'd be in favour of deprecating this, it would force us to use our own API.

Reply:
#1175